### PR TITLE
[3.8.0]add fbx lod template

### PR DESCRIPTION
**Pull request by naturalch**
Re: https://github.com/cocos/3d-tasks/issues/15647

### Changelog

*

-------

### Continuous Integration

This pull request:

* [ ] needs automatic test cases check.
  > Manual trigger with `@cocos-robot run test cases` afterward.
* [x] does not change any runtime related code or build configuration
  > If any reviewer thinks the CI checks are needed, please uncheck this option, then close and reopen the issue.

-------

### Compatibility Check

This pull request:

* [ ] changes public API, and have ensured backward compatibility with [deprecated features](https://github.com/cocos/cocos-engine/blob/v3.5.0/docs/contribution/deprecated-features.md).
* [ ] affects platform compatibility, e.g. system version, browser version, platform sdk version, platform toolchain, language version, hardware compatibility etc.
* [ ] affects file structure of the build package or build configuration which requires user project upgrade.
* [ ] **introduces breaking changes**, please list all changes, affected features and the scope of violation.

### CODE REVIEW

File: editor/inspector/assets/fbx/model.js [hunk 1]
```diff
@@ -121,6 +125,17 @@ exports.template = /* html */`
             </ui-prop>
         </div>
     </ui-section>
+    <ui-section class="lods config" cache-expand="fbx-mode-lods">
+        <div slot="header" class="lods-header">
+            <ui-checkbox slot="content" class="lods-checkbox"></ui-checkbox>
```
Comment (arsen2010) [inline editor/inspector/assets/fbx/model.js]: `slot="content"`  is useless

File: editor/inspector/assets/fbx/model.js [hunk 2]
```diff
@@ -121,6 +125,17 @@ exports.template = /* html */`
...
+            <ui-label value="LODS" tooltip="To import LODs, please make sure the LOD mesh names are ending with _LOD#"></ui-label>
```
Comment (arsen2010) [inline editor/inspector/assets/fbx/model.js:127]: `tooltip`  need i18n

File: editor/inspector/assets/fbx/model.js [hunk 3]
```diff
@@ -121,6 +125,17 @@ exports.template = /* html */`
...
+        </div>
+        <div class="lod-items"></div>
+        <div class="no-lod-label" hidden>There is no LOD(Level of Details) group can be detected in this model.LOD levels can be automatically generated with above settings.</div>
```
Comment (arsen2010) [inline editor/inspector/assets/fbx/model.js:130]: The innerText need to be i18n words too.

File: editor/inspector/assets/fbx/model.js [hunk 4]
```diff
@@ -152,6 +168,97 @@ ui-section {
...
+.lods .load-mask {
+    position: absolute;
+    width: 100%;
+    height: 100%;
+    top: 0;
+    left: 0;
+    text-align: center;
+    background-color: #1b1d1db0;
+    z-index: 10;
+    display: flex;
```
Comment (arsen2010) [inline editor/inspector/assets/fbx/model.js]: duplicated below

File: editor/inspector/assets/fbx/model.js [hunk 5]
```diff
@@ -553,6 +666,23 @@ const Elements = {
             updateElementReadonly.call(panel, panel.$.meshOptimizerTargetRatioSlider);
         },
     },
+    enableSmartLink: {
```
Comment (arsen2010) [inline editor/inspector/assets/fbx/model.js]: These codes should be not belong to this PR

File: editor/inspector/assets/fbx/model.js [hunk 6]
```diff
@@ -669,9 +894,14 @@ exports.methods = {
         this.dispatch('change');
         this.dispatch('track', { tab: 'model', prop, value: event.target.value });
     },
+    apply() {
+        this.$.loadMask.style.display = 'flex';
+    }
 };
 
 exports.ready = function() {
+    Editor.Message.addBroadcastListener('fbx-inspector:apply', this.apply.bind(this));
```
Comment (arsen2010) [inline editor/inspector/assets/fbx/model.js]: Need to cache a variable equal to `this.apply.bind(this)` , and use it on close period

File: editor/inspector/assets/fbx/model.js [hunk 7]
```diff
@@ -695,10 +925,55 @@ exports.update = function(assetList, metaList) {
 };
 
 exports.close = function() {
+    Editor.Message.removeBroadcastListener('fbx-inspector:apply', this.apply.bind(this));
```
Comment (arsen2010) [inline editor/inspector/assets/fbx/model.js]: see above

File: editor/inspector/assets/fbx/model.js [hunk 8]
```diff
@@ -695,10 +925,55 @@ exports.update = function(assetList, metaList) {
...
+                    value="${Editor.Utils.Math.multi(lodItem.faceCount, 100)}"
+                    ${ isBuiltin ? 'disabled' : '' }>
+                </ui-num-input>
+            </div>
+            <div class="right">
+                <div class="triangles">
+                    <span> ${lodItem.triangleCount} Triangles</span>
+                </div>
+                <div class="operator" ${ isBuiltin ? 'hidden' : '' }>
+                    <ui-icon value="add" key="${index}" path="insertLod" tooltip="insert after this LOD"></ui-icon>
```
Comment (arsen2010) [inline editor/inspector/assets/fbx/model.js:963]: `tooltip` need to be 1i8n key, below too

File: editor/inspector/assets/fbx/model.js [hunk 9]
```diff
@@ -656,7 +786,102 @@ const Elements = {
...
+                }
+            });
+            // Listening to the addition and removal of the lod hierarchy
+            panel.$.lodItems.addEventListener('click', (event) => {
+                event.stopPropagation();
+                event.preventDefault();
+                const path = event.target.getAttribute('path');
+                const index = Number(event.target.getAttribute('key'));
+                if (path === 'insertLod') {
+                    if (Object.keys(panel.meta.userData.lods.options).length >= 8) {
```
Comment (arsen2010) [inline editor/inspector/assets/fbx/model.js]: suggest new a variable to 'panel.meta.userData.lods' for below codes

File: editor/inspector/assets/fbx/model.js [hunk 10]
```diff
@@ -656,7 +786,102 @@ const Elements = {
...
+                    const nextScreenRatio = panel.meta.userData.lods.options[index + 1] ? panel.meta.userData.lods.options[index + 1].screenRatio : 0;
+                    const preFaceCount = panel.meta.userData.lods.options[index].faceCount;
+                    const nextFaceCount = panel.meta.userData.lods.options[index + 1] ? panel.meta.userData.lods.options[index + 1].faceCount : 0;
+                    const option = {
+                        screenRatio: (preScreenRatio + nextScreenRatio) / 2,
+                        triangleCount: 0,
+                        faceCount: (preFaceCount + nextFaceCount) / 2,
+                    };
+                    // Insert the specified lod level
+                    for (let key = Object.keys(panel.meta.userData.lods.options).length - 1; key > index; key--) {
```
Comment (arsen2010) [inline editor/inspector/assets/fbx/model.js]: it is not 'key' , it is keyIndex

File: editor/inspector/assets/fbx/model.js [hunk 11]
```diff
@@ -656,7 +786,102 @@ const Elements = {
...
+                        panel.meta.userData.lods.options[parseInt(key) + 1] = panel.meta.userData.lods.options[key];
```
Comment (arsen2010) [inline editor/inspector/assets/fbx/model.js]: parseInt useless, because it is keyIndex originally

File: editor/inspector/assets/fbx/model.js [hunk 12]
```diff
@@ -656,7 +786,102 @@ const Elements = {
...
+                    }
+                    panel.meta.userData.lods.options[index + 1] = option;
+                    // update panel
+                    panel.$.lodItems.innerHTML = getLodItemHTML(panel.meta.userData.lods.options, panel.meta.userData.lods.isBuiltin);
```
Comment (arsen2010) [inline editor/inspector/assets/fbx/model.js]: when data change ，can call existed method. 
`Elements.lods.update.call(panel);`

File: editor/inspector/assets/fbx/model.js [hunk 13]
```diff
@@ -656,7 +786,102 @@ const Elements = {
...
+                        return;
+                    }
+                    // Delete the specified lod level
+                    for (let key = index; key < Object.keys(panel.meta.userData.lods.options).length; key++) {
+                        panel.meta.userData.lods.options[key] = panel.meta.userData.lods.options[key + 1];
+                    }
+                    delete panel.meta.userData.lods.options[Object.keys(panel.meta.userData.lods.options).length - 1];
+                    // update panel
+                    panel.$.lodItems.innerHTML = getLodItemHTML(panel.meta.userData.lods.options, panel.meta.userData.lods.isBuiltin);
+                    panel.dispatch('change');
```
Comment (arsen2010) [inline editor/inspector/assets/fbx/model.js:847]: this method call from click , so after dispatch chane , need add `panel.dispatch('snapshot');`

File: editor/inspector/assets/fbx/model.js (hunk 2 above)
Comment (naturalch) [reply editor/inspector/assets/fbx/model.js:127]: currently just use the English

File: editor/inspector/assets/fbx/model.js (hunk 3 above)
Comment (naturalch) [reply editor/inspector/assets/fbx/model.js:130]: currently just use the English too

File: editor/inspector/assets/fbx/model.js (hunk 13 above)
Comment (naturalch) [reply editor/inspector/assets/fbx/model.js:847]: You are right

File: editor/inspector/assets/fbx/model.js (hunk 6 above)
Comment (naturalch) [reply editor/inspector/assets/fbx/model.js]: you are right. `bind()` wil return a new function each time

File: editor/inspector/assets/fbx/model.js (hunk 8 above)
Comment (naturalch) [reply editor/inspector/assets/fbx/model.js:963]: currently just use the English too

**Review by arsen2010 — APPROVED** [review]

**Review by arsen2010 — APPROVED** [review]

**Review by VisualSJ — APPROVED** [review]
